ui/src/app/views/cohort-edit/component.spec.ts

```diff
@@ -0,0 +1,262 @@
+      setTimeout(() => {
+        const workspaceMatch = this.getMatchingWorkspaceOrSendError(wsNamespace, wsId, observer);
+        if (workspaceMatch !== undefined) {
+          const index = workspaceMatch.cohorts.findIndex(function(cohort: Cohort) {
+            if (cohort.id === cId) {
+              return true;
+            }
+            return false;
+          });
+          if (index !== -1) {
+            workspaceMatch.cohorts[index] = newCohort;
+            observer.complete();
+          } else {
+            observer.error(new Error(`Error updating. No cohort with id: ${cId} \
+              exists in workspace ${wsNamespace}, ${wsId} in cohort service stub`));
+          }
+        }
+      }, 0);
+    });
+    return observable;
+  }
+
+  public createCohort(wsNamespace: string,
+                      wsId: string, newCohort: Cohort): Observable<Cohort> {
+    const observable = new Observable(observer => {
+      setTimeout(() => {
+        const workspaceMatch = this.getMatchingWorkspaceOrSendError(wsNamespace, wsId, observer);
+        if (workspaceMatch !== undefined) {
+          observer.next(workspaceMatch.cohorts.find(function(cohort: Cohort) {
+            if (cohort.id === newCohort.id) {
+              observer.error(new Error(`Error creating. Cohort with \
+id: ${cohort.id} already exists.`));
+              return true;
+            }
+          }));
+          workspaceMatch.cohorts.push(newCohort);
+          observer.complete();
+        }
+      }, 0);
+    });
+    return observable;
+  }
+
+  public getCohortsInWorkspace(wsNamespace: string,
+                               wsId: string): Observable<CohortListResponse> {
+    const observable = new Observable(observer => {
+      setTimeout(() => {
+        const workspaceMatch = this.getMatchingWorkspaceOrSendError(wsNamespace, wsId, observer);
+        if (workspaceMatch !== undefined) {
+          observer.next({items: workspaceMatch.cohorts});
+          observer.complete();
+        }
+      }, 0);
+    });
+    return observable;
+  }
+}
+
+const activatedRouteStub  = {
+  snapshot: {
+    url: [
+      {path: 'workspace'},
+      {path: WorkspaceComponent.DEFAULT_WORKSPACE_NS},
+      {path: WorkspaceComponent.DEFAULT_WORKSPACE_ID},
+      {path: 'cohorts'},
+      {path: 'create'}
+    ]
+  }
+};
+
+describe('CohortEditComponent', () => {
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        RouterTestingModule,
+        FormsModule
+      ],
+      declarations: [
+        CohortEditComponent
+      ],
+      providers: [
+        { provide: CohortsService, useValue: new CohortsServiceStub() },
+        { provide: ActivatedRoute, useValue: activatedRouteStub }
+      ] }).compileComponents();
+  }));
+
+
+
+  it('displays blank input fields when creating a new cohort', async(() => {
+    const context = new Context(TestBed);
+    context.fixture.detectChanges();
+    setTimeout(function(){
```

Comment:
In this case, it doesn't actually need either. Here, we can't use tick() though, because its an async() test rather than a fakeAsync()